### ISSUE TITLE
Any chance of implementation of L1 distance?

### ISSUE THREAD

**Issue by Slasher111** [nmslib/hnswlib#349]
Hello and thank you for the amazing library I'm widely using in my projects!
Do you have any plans of  adding L1 as one of the distance functions?
I know that nmslib has L1, but it also has double memory consumption compared to hnswlib :)

**Comment by yurymalkov** [comment]
Hi. Currently, there are no plans to add it. But it is pretty straightforward to implement if it is needed (by copying it from nmslib)

**Comment by Slasher111** [comment]
It is not _that_ straightforward for me, since I'm not that good in C++ :). But, as a "quick dirty hack" (https://github.com/nmslib/hnswlib/issues/256) I suppose that can be done via changing functions in https://github.com/nmslib/hnswlib/blob/master/hnswlib/space_l2.h 
into necessary functions that are in https://github.com/nmslib/nmslib/blob/master/similarity_search/src/distcomp_lp.cc.
Am I right? Or it is far beyond that and I should not even try? :)

**Comment by yurymalkov** [comment]
Yeah. As a dirty hack it should be good. I guess just changing https://github.com/nmslib/hnswlib/blob/1866a1df7961c42cd4efb0c8ffc665d6209447f9/hnswlib/space_l2.h#L17 to abs instead of square here and removing all of the contents in `#if`https://github.com/nmslib/hnswlib/blob/1866a1df7961c42cd4efb0c8ffc665d6209447f9/hnswlib/space_l2.h#L177  (so that only that code will be used) should do the trick.
This should be as fast as manually optimized version for an intel C++ compiler. I guess there will be some hit of performance for gcc, but not really sure.

**Comment by Slasher111** [comment]
And it worked out, thank you very much! I couldn't make an exact performance test yet, but it feels like no slowdown occured. Now I have my lovely L1 distance! :)
